Document how to enable parallel step exec for all steps (#3605)
woodpecker-ci/woodpecker: docs/docs/20-usage/20-workflow-syntax.md

## Steps

Every step of your workflow executes commands inside a specified container.<br>

> The defined commands are executed serially by default, if they should run in parallel use [`depends_on`](./20-workflow-syntax.md#depends_on).<br>

Reviewer: ```suggestion
The defined steps are executed in sequence by default, if they should run in parallel you can use [`depends_on`](./20-workflow-syntax.md#depends_on).<br>
```

       - go test
```

:::note

> Add a single empty `depends_on` execute all steps in parallel:

Reviewer: Maybe like this:

```suggestion
You can define a step to start immediately without dependencies by adding an empty `depends_on: []`. By setting `depends_on` on a single step all other steps will be immediately executed as well if no further dependencies are specified. 
```
